config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -137,21 +149,35 @@ private Element createDeleteBundle(final Document document, List<Entity> entitie
         return bundleDocumentBuilder.build(document, deleteBundleEntities);
     }
 
+    /**
+     * Creates the DELETE environment bundle element.
+     *
+     * @param document Document
+     * @param entities Entities packaged in the deployment bundle
+     * @return Delete bundle Element for the Annotated Bundle
+     */
+    private Element createDeleteEnvBundle(final Document document, List<Entity> entities) {
+        List<Entity> filteredEntities = copyFilteredEntitiesForDeleteBundle(entities, FILTER_ENV_ENTITIES);
+        filteredEntities.forEach(e -> e.setMappingAction(MappingActions.DELETE));
+        return bundleDocumentBuilder.build(document, filteredEntities);
+    }
+
     /**
      * Copies all the filtered entities in the reverse order to a new {@link List}. The entries in the DELETE bundle
      * must be in the reverse order of deployment bundle.
-     *
+     * <p>
      * All Environment entities and Folders are skipped from the DELETE bundle list.
      *
-     * @param entities Entities in the deployment bundle
+     * @param entities        Entities in the deployment bundle
+     * @param entityPredicate predicate for entity inclusion
      * @return Filtered list of entities in the reverse order
      */
-    private List<Entity> copyFilteredEntitiesForDeleteBundle(List<Entity> entities) {
+    private List<Entity> copyFilteredEntitiesForDeleteBundle(List<Entity> entities, Predicate<Entity> entityPredicate) {
```

Comment:
Can you change the var name to `filter` maybe?

Reply:
done